src/object/srv_ec_aggregate.c

```diff
@@ -1587,23 +1601,33 @@ agg_process_holes(struct ec_agg_entry *entry)
 		rc = dss_abterr2der(rc);
 		goto ev_out;
 	}
-
+#if HUH
+	if (*status != 0) {
+		rc = *status;
+		goto ev_out;
+	}
+#endif
 	/* Update local vos with replicate */
 	iod.iod_name = entry->ae_akey;
 	iod.iod_type = DAOS_IOD_ARRAY;
 	iod.iod_size = entry->ae_rsize;
 	iod.iod_nr = stripe_ud.asu_cell_cnt;
 	iod.iod_recxs = stripe_ud.asu_recxs;
+
 	entry->ae_sgl.sg_nr = 1;
-	/* write the reps to vos */
-	agg_param = container_of(entry, struct ec_agg_param, ap_agg_entry);
-	rc = vos_obj_update(agg_param->ap_cont_handle, entry->ae_oid,
-			    entry->ae_cur_stripe.as_hi_epoch, 0, 0,
-			    &entry->ae_dkey, 1, &iod, NULL,
-			    &entry->ae_sgl);
-	if (rc) {
-		D_ERROR("vos_update_begin failed: "DF_RC"\n", DP_RC(rc));
-		goto ev_out;
+	agg_param = container_of(entry, struct ec_agg_param,
+				 ap_agg_entry);
+	if (iod.iod_nr) {
```

Comment:
if iod_nr == 0, does it still need to delete the parity?  Just curious, when asu_cell_cnt can be 0 here?

Reply:
Good catch.  The asu_cell_cnt will be zero if all of the hole records lying within the stripe are older than the parity (e.g., punch at an epoch, full stripe write at a later epoch, then EC aggregation runs).  In this case, should not delete the parity.